cpp/gtp.cpp

```diff
@@ -1029,67 +1030,126 @@ int MainCmds::gtp(int argc, const char* const* argv) {
         }
       }
     }
-
-    else if(command == "time_settings") {
-      double mainTime;
-      double byoYomiTime;
-      int byoYomiStones;
-      if(pieces.size() != 3
-         || !Global::tryStringToDouble(pieces[0],mainTime)
-         || !Global::tryStringToDouble(pieces[1],byoYomiTime)
-         || !Global::tryStringToInt(pieces[2],byoYomiStones)
-         ) {
-        responseIsError = true;
-        response = "Expected 2 floats and an int for time_settings but got '" + Global::concat(pieces," ") + "'";
-      }
-      else if(isnan(mainTime) || mainTime < 0.0 || mainTime > 1e50) {
-        responseIsError = true;
-        response = "invalid main_time";
-      }
-      else if(isnan(byoYomiTime) || byoYomiTime < 0.0 || byoYomiTime > 1e50) {
-        responseIsError = true;
-        response = "invalid byo_yomi_time";
-      }
-      else if(byoYomiStones < 0 || byoYomiStones > 100000) {
-        responseIsError = true;
-        response = "invalid byo_yomi_stones";
-      }
-      else {
-        TimeControls tc;
-        //This means no time limits, according to gtp spec
-        if(byoYomiStones == 0 && byoYomiTime > 0.0) {
-          //do nothing, tc already no limits by default
-        }
-        //Absolute time
-        else if(byoYomiStones == 0) {
-          tc.originalMainTime = mainTime;
-          tc.increment = 0.0;
-          tc.originalNumPeriods = 0;
-          tc.numStonesPerPeriod = 0;
-          tc.perPeriodTime = 0.0;
-          tc.mainTimeLeft = mainTime;
-          tc.inOvertime = false;
-          tc.numPeriodsLeftIncludingCurrent = 0;
-          tc.numStonesLeftInPeriod = 0;
-          tc.timeLeftInPeriod = 0;
-        }
-        else {
-          tc.originalMainTime = mainTime;
-          tc.increment = 0.0;
-          tc.originalNumPeriods = 1;
-          tc.numStonesPerPeriod = byoYomiStones;
-          tc.perPeriodTime = byoYomiTime;
-          tc.mainTimeLeft = mainTime;
-          tc.inOvertime = false;
-          tc.numPeriodsLeftIncludingCurrent = 1;
-          tc.numStonesLeftInPeriod = 0;
-          tc.timeLeftInPeriod = 0;
-        }
-
-        engine->bTimeControls = tc;
-        engine->wTimeControls = tc;
-      }
-    }
+	else if (command == "kgs-time_settings" && pieces[0] != "canadian") {
+		double mainTime;
+		double byoYomiTime;
+		int originalNumPeriods;
+		if (pieces.size() != 4
+			|| pieces[0] != "byoyomi"
+			|| !Global::tryStringToDouble(pieces[1], mainTime)
+			|| !Global::tryStringToDouble(pieces[2], byoYomiTime)
+			|| !Global::tryStringToInt(pieces[3], originalNumPeriods)
+			) {
+			if (pieces.size() != 2
+				|| pieces[0] != "absolut"
```

Comment:
Is "absolut" a typo?

Reply:
Oops, i confess i did'nt test "absolute". It's a typo, i looked it up in the Doku (kgsGtp.xhtml)

Reply:
Dokumentation for kgsgtp: http://www.weddslist.com/kgs/how/kgsGtp.html

Reply:
by the way, thx for all the excellent work (hurray!)